DataDog/ecommerce-workshop
# Update RUM, add session replay, update GKE deploy

This will update the RUM package being used in the `ddtraining/storefront-fixed` and `ddtraining/storefront` images, which will also enable session replay. I've tested locally and can confirm that RUM data + session replay are being populated in the DD app but some more eyes on this would be helpful as well.

In the RUM init script, I've also updated it so the `service` property inherits from the `DD_SERVICE` env variable set in the deploy config, rather than being hardcoded to `storedog-ui`. 

This is needed to resolve #177 . After this is approved, I'll do a minor release for 2.1.2 and update the GKE yaml accordingly.

**How to test**
1. Pull down branch locally
2. Follow [these instructions](https://github.com/DataDog/ecommerce-workshop/blob/main/development.md) to build the storefront code with the new patches
3. Setup a new RUM app in DD and get the client_token and application_id
4. Run storedog locally using this command:
```
POSTGRES_USER=postgres POSTGRES_PASSWORD=postgres DD_API_KEY=[API KEY] DD_APPLICATION_ID=[RUM APP ID] DD_CLIENT_TOKEN=[RUM CLIENT TOKEN] docker compose -f deploy/docker-compose/docker-compose-fixed-instrumented.yml up
```
5. Click through storedog to create some session actions
6. Verify that RUM data is being pulled through and that session replays are available

**Notes**
There may be some labs out there that are built upon RUM use, which assume storedog is using the older RUM package. I'm unsure how to verify if this is true
